### PR TITLE
Adding support for using a provided ArrayBuffer as the heap.

### PR DESCRIPTION
**Pull request by flagxor**
Take a 3rd parameter to initiateModule that takes an ArrayBuffer
to use as the heap for the module.

### CODE REVIEW

**Comment by aseemgarg89** [comment]
looks good to me.

**Comment by JSStats** [comment]
Where does this sit wrt aligning the memory base on a page boundary, which is an expected requirement for wasm?

File: test/mjsunit/wasm/outer-memory.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// Copyright 2015 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
```
Comment (titzer) [inline test/mjsunit/wasm/outer-memory.js]: Can we make this test part of the other memory test? The reason is that it was quite painful to update each test after renumbering opcodes (which will happen again in a little while).

**Comment by titzer** [comment]
lgtm other than the new test being in a new file.

**Comment by flagxor** [comment]
done

**Pull request merged**